course/frontend/prototypes/scripts/main.js

```diff
@@ -20,4 +20,45 @@ console.log(1, 'fiestaD.brand should be ford', fiestaD.brand);
 console.log(2, 'fiestaD.model should be fiesta', fiestaD.model);
 console.log(3, 'fiestaD.motor should be diesel', fiestaD.motor);
 console.log(4, 'fiestaD.start() should start diesel motor', fiestaD.start());
-console.log(4, 'fiestaD.stop() should stop diesel motor', fiestaD.stop());
\ No newline at end of file
+console.log(4, 'fiestaD.stop() should stop diesel motor', fiestaD.stop());
+
+//CHALLENGE PROTOTYPE
+//Person
+function Person (firstName, lastName, age, gender, interests){
+    this.firstName = firstName
+    this.lastName = lastName
+    this.age = age
+    this.gender = gender
+    this.interests = interests
+}
+Person.prototype.greeting = function () {
+    return `Hi i'm ${this.firstName}`
+}
+Person.prototype.farewell = function () {
+    return `${this.firstName} has left the building.
+    Bye for now!`
+}
+Person.prototype.bio = function () {
+    return `${this.firstName} ${this.lastName}is ${this.age} years old. ${(this.gender == 'famel') ? "She" : "He"} likes ${this.interests.join(", ")}`;
+}
+//Student
+function Student(firstName, lastName, age, gender, interests) {
+    Person.call(this,firstName, lastName, age, gender, interests)
+}
+Student.prototype = new Person()
```

Comment:
Esta línea no es necesaria??? Sin ella funciona igual...

Reply:
hey @AsierDev, Hemos de poner `{Student o Teacher}.prototype = new Person()` para que hereden los metodos de Person!

Reply:
Ah ok! Justo lo estabamos comentando antes despues de clase y ninguno lo teniamos claro... Thanks!

Reply:
ya veras! prueba sin poner esas lineas, y cuando intentas acceder a los metodos de su padre (Person) no existen! 🐶

Reply:
Yo también lo entiendo como @VGamezz19

Reply:
Yo he tenido que añadir 
`function Teacher(firstName, lastName, age, gender, interests, subject) {
    Person.call(this, firstName, lastName, age, gender, interests)
}

Teacher.prototype.constructor = Object.create(Person.prototype);
Teacher.prototype.constructor = Teacher;`

para que me herede de Person, pero a su vez cree un objeto propio

Reply:
ah vale, que esto funciona como un markdown

Reply:
Yo he resuelto mi función Teacher tal que así;
function Teacher(firstName,lastName,age,gender,interests,subject){
	Teacher.prototype.subject = function(){
		return subject;
	}
	Person.call(this,firstName,lastName,age,gender,interests,subject);
}
Teacher.prototype = new Person();

Reply:
@Monty4 Me parece correcto! pero dentro del constructor de `Teacher` se ha de definir el atributo `subject` con **this**, poniendo `Teacher.prototype.subject` haces que todos los nuevos `Objectos` que cree de tipo `Teacher` compartiran Subject